packages/adapter-planetscale/src/planetscale.ts

```diff
@@ -193,3 +178,32 @@ const adapter = new PrismaPlanetScale(client)
     })
   }
 }
+
+export class PrismaPlanetScale extends PlanetScaleQueryable<planetScale.Client> implements DriverAdapter {
+  constructor(client: planetScale.Client) {
+    // this used to be a check for constructor name at same point (more reliable when having multiple copies
+    // of @planetscale/database), but that did not work with minifiers, so we reverted back to `instanceof`
+    if (!(client instanceof planetScale.Client)) {
+      throw new TypeError(`PrismaPlanetScale must be initialized with an instance of Client:
+import { Client } from '@planetscale/database'
+const client = new Client({ url })
+const adapter = new PrismaPlanetScale(client)
+`)
+    }
+    super(client)
+  }
+
+  getConnectionInfo(): Result<ConnectionInfo> {
+    const url = this.client.connection()['url'] as string
+    const dbName = new URL(url).pathname.slice(1) /* slice out forward slash */
+    return ok({
+      schemaName: dbName,
+    })
+  }
+
+  async transactionContext(): Promise<Result<TransactionContext>> {
+    const conn = this.client.connection()
+    const ctx = new PlanetScaleTransactionContext(conn)
+    return Promise.resolve(ok(ctx))
```

Comment:
```suggestion
    return ok(ctx)
```
it's already an `async` function

Reply:
Not quite:

<img width="934" alt="Screenshot 2024-09-03 at 12 20 00" src="https://github.com/user-attachments/assets/12626e36-52a3-492a-8469-2aa27d76bd51">

Reply:
Then it's a problem with the eslint rule being wrong and not taking the implemented interfaces into account 😆
It's fine though, let's leave `Promise.resolve` to shut it up then.

Reply:
You did `  // eslint-disable-next-line @typescript-eslint/require-await` in other places in this PR to solve the same problem though.

Reply:
I've looked more into it. I wasn't the original person who used `// eslint-disable-next-line @typescript-eslint/require-await` (Joel did). Afaik, this rule can be safely disabled as we're not using generator functions (`yield` behaves differently depending on whether a Promise is explicitly returned afaik, that's why this rule exists in the first place).

I will disable it at the top of the files, rather than once for each of these methods.

Reply:
Sounds good!